certora/specs/SummaryLib.spec

```diff
@@ -86,18 +66,19 @@ function mockPrice() returns uint256 {
     return updatedPrice;
 }
 
-
 // Ensure this function is only used when no interest is accrued, or enforce that the last update matches the current timestamp.
 function positionAsAssets (address borrower) returns (uint256, uint256) {
-    PreLiquidation.Market m = MORPHO.market_(currentContract.ID);
-    PreLiquidation.Position p = MORPHO.position_(currentContract.ID, borrower);
+    uint256 borrowerShares = MORPHO.borrowShares(currentContract.ID, borrower);
+    uint256 borrowerCollateral = MORPHO.collateral(currentContract.ID, borrower);
 
-    uint256 collateralQuoted = require_uint256(summaryMulDivDown(p.collateral, mockPrice(), ORACLE_PRICE_SCALE()));
+    uint256 collateralQuoted = summaryMulDivDown(borrowerCollateral, mockPrice(), ORACLE_PRICE_SCALE());
 
     // Safe require because the implementation would revert, see rule zeroCollateralQuotedReverts.
     require collateralQuoted > 0;
 
-    uint256 borrowed = require_uint256(summaryToAssetsUp(p.borrowShares, m.totalBorrowAssets, m.totalBorrowShares));
+    uint256 totalAssets = MORPHO.virtualTotalBorrowAssets(currentContract.ID);
+    uint256 totalShares = MORPHO.virtualTotalBorrowShares(currentContract.ID);
```

Comment:
Same, these operations can't overflow.

Reply:
Are you suggesting to define `totalAssets` and `totalShares` as `mathint` ? I think it would make it more complicated because we would then need to `assert_uint256` on those to use them in the next computation with `summaryMulDivUp`